src/utils/collect.ts

```diff
@@ -0,0 +1,11 @@
+// Combines filter + map in a single pass. Inspired by Scala's collect
+export const collect = <A, B>(arr: A[], f: (a: A) => B | undefined): B[] => {
```

Comment:
Oh interesting! I'd not used `collect` in Scala before. The name surprised me for a moment, in Ruby [`collect` is an alias for `map`](https://ruby-doc.org/core-2.4.1/Array.html#method-i-map) (or maybe it's the other way around).

Reply:
Perhaps `filterMap` would be a better name!